File: Stephannie/Exercise-12.py
#Love calculator

name1 = input("Enter your name: ").lower()
name2 = input("Enter your partner's name: ").lower()

combined_name = name1 + name2
combined_name_length = len(combined_name)
num_of_T = combined_name.count("t")
num_of_R = combined_name.count("r")
num_of_U = combined_name.count("u")
num_of_E = combined_name.count("e")

num_of_L = combined_name.count("l")
num_of_O = combined_name.count("o")
num_of_V = combined_name.count("v")
num_of_E2 = combined_name.count("e")

total_true = num_of_T + num_of_R + num_of_U + num_of_E
total_love = num_of_L + num_of_O + num_of_V + num_of_E2
Score = total_love + total_true

percentage_score = (Score / combined_name_length) * 100

if percentage_score < 10 or percentage_score > 90:
    print(f"Your score is {Score}, you go together like coke and mentos.")
elif percentage_score >= 40 and percentage_score <= 50:
    print(f"Your score is {Score}, you are alright together.")
else:
    print(f"Your score is {Score}.")

File: YaroWilliams/Exercise12.java
import java.util.Scanner;
public class Exercise12 {
    public static void  main(String[] args){
        Scanner input = new Scanner(System.in);

        System.out.println("Hey there, Welcome to the love cirlcle, we calculate the love compatibility between you and your patner!");
        System.out.println(" ");

        System.out.println("Enter the first name:");
        String name1 = input.nextLine().toLowerCase();

        System.out.println("Enter the second name:");
        String name2 = input.nextLine().toLowerCase();

        String combinedNames = name1 + name2;

        int trueCount = countMatches(combinedNames, 't')+
                        countMatches(combinedNames, 'r')+
                        countMatches(combinedNames, 'u')+
                        countMatches(combinedNames, 'e');

        int loveCount = countMatches(combinedNames, 'l')+
                        countMatches(combinedNames, 'o')+
                        countMatches(combinedNames, 'v')+
                        countMatches(combinedNames, 'e');

        int compatScore = (trueCount + loveCount) * 10;

        if(compatScore < 10 || compatScore > 90){
            System.out.println("Your love compatibility is "+compatScore+"%. You go together like coke and mentos.");
        }else if(compatScore >=40 && compatScore<= 50){
            System.out.println("Your love compatibility is "+compatScore+"%. You are alright together.");
        }else{
            System.out.println("Your love compatibility is "+compatScore+"%");
        }
        input.close();
    }
    public static int countMatches(String text1, char letter){
        int count = 0;
        for(char i: text1.toCharArray()){
            if(i == letter){
                count++;
            }
        }
        return count;
    }    
}

